foundry/pixi/containers/canvasLayers/placeablesLayers/wallsLayer.d.ts

```diff
@@ -0,0 +1,241 @@
+    walls: Wall[],
+    {
+      bounds,
+      blockMovement,
+      blockSenses
+    }?: { bounds?: Rectangle | null; blockMovement?: boolean; blockSenses?: boolean }
+  ): PointArray[];
+
+  /**
+   * Test whether movement along a given Ray collides with a Wall.
+   * @param ray - The attempted movement
+   * @returns Does a collision occur?
+   */
+  checkCollision(
+    ray: Ray,
+    {
+      blockMovement,
+      blockSenses,
+      mode
+    }?: { blockMovement?: boolean; blockSenses?: boolean; mode?: 'all' | 'closest' | 'any' }
+  ): boolean;
+
+  /**
+   * Highlight the endpoints of Wall segments which are currently group-controlled on the Walls layer
+   */
+  highlightControlledSegments(): void;
+
+  /** @override */
+  releaseAll(): number;
+
+  /** @override */
+  pasteObjects(position: Point, options?: {}): Promise<Wall[]>;
+
+  /**
+   * Pan the canvas view when the cursor position gets close to the edge of the frame
+   * @param event - The originating mouse movement event
+   * @param x     - The x-coordinate
+   * @param y     - The y-coordinate
+   */
+  protected _panCanvasEdge(event: MouseEvent, x: number, y: number): void | Promise<void>;
+
+  /**
+   * Get the endpoint coordinates for a wall placement, snapping to grid at a specified precision
+   * Require snap-to-grid until a redesign of the wall chaining system can occur.
+   * @param  point - The initial candidate point
+   * @param  snap  - Whether to snap to grid
+   *                 (default: `true`)
+   * @returns The endpoint coordinates [x,y]
+   */
+  protected _getWallEndpointCoordinates(point: Point, { snap }?: { snap?: boolean }): PointArray;
+
+  /**
+   * The Scene Controls tools provide several different types of prototypical Walls to choose from
+   * This method helps to translate each tool into a default wall data configuration for that type
+   * @param tool - The active canvas tool
+   */
+  protected _getWallDataFromActiveTool(
+    tool: string
+  ):
+    | {
+        move: ValueOf<typeof CONST['WALL_MOVEMENT_TYPES']>;
+        sense: ValueOf<typeof CONST['WALL_SENSE_TYPES']>;
+        door: ValueOf<typeof CONST['WALL_DOOR_TYPES']>;
+      }
+    | Wall['data'];
+
+  /** @override */
+  protected _onClickLeft(event: PIXI.InteractionEvent): void;
+
+  /** @override */
+  protected _onDragLeftStart(event: PIXI.InteractionEvent): void;
+
+  /** @override */
+  protected _onDragLeftMove(event: PIXI.InteractionEvent): void;
+
+  /** @override */
+  protected _onDragLeftDrop(event: PIXI.InteractionEvent): void;
+
+  /** @override */
+  protected _onDragLeftCancel(event: PointerEvent): void;
+
+  /** @override */
+  protected _onClickRight(event: PIXI.InteractionEvent): void;
+
+  /**
+   * Test a single Ray against a single Wall
+   * @param ray  - The Ray being cast
+   * @param wall - The Wall against which to test
+   * @returns An intersection, if one occurred
+   */
+  static testWall(ray: Ray, wall: Wall): Vector2 | null;
```

Comment:
I think `Vector2` is being replaced in #534, so please coordinate.

Reply:
RayIntersection was the correct type.